src/citrine/ara/columns.py

```diff
@@ -0,0 +1,83 @@
+"""Column definitions for Ara."""
+from typing import Type, Optional, List  # noqa: F401
+from abc import abstractmethod
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization import properties
+
+
+class Column(PolymorphicSerializable['Column']):
+    """A column in the Ara table, defined as some operation on a variable.
+
+    Abstract type that returns the proper type given a serialized dict.
+    """
+
+    @abstractmethod
+    def _attrs(self) -> List[str]:
+        pass  # pragma: no cover
+
+    def __eq__(self, other):
+        try:
+            return all([
+                self.__getattribute__(key) == other.__getattribute__(key) for key in self._attrs()
+            ])
+        except AttributeError:
+            return False
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        types: List[Type[Serializable]] = [
+            RealMeanColumn, IdentityColumn
+        ]
+        return next(x for x in types if x.type == data["type"])
+
+
+class RealMeanColumn(Serializable['RealMeanColumn'], Column):
+    """Column containing the mean of a real-valued variable.
+
+    Parameters
+    ----------
+    data_source: str
+        short_name of the variable to use when populating the column
```

Comment:
Do all columns have a `data_source` field and do all variables have a `short_name` field? This seems to be implying that there is a uniqueness requirement (all variables in a given Ara definition must have unique short names). Is that true, and what happens if that requirement is not met?

Reply:
They do.  I made a comment about their interfaces in response to @kroenlein and opened #166.  I've added constructor-time validations in AraDefinition that `short_name` is unique across the variables and each `data_source` is present as the `short_name` of one of the variables.